extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/cache/SynchronousInfinispanGet.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.infinispan.client.runtime.cache;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class SynchronousInfinispanGet {
+    Map<String, Map<Object, CompletableFuture<Object>>> synchronousGetLocks = new ConcurrentHashMap<>();
+
+    public void initIfAbsent(String cacheName) {
+        synchronousGetLocks.putIfAbsent(cacheName, new ConcurrentHashMap<>());
+    }
+
+    public Map<Object, CompletableFuture<Object>> get(String cacheName) {
+        return synchronousGetLocks.get(cacheName);
+    }
```

Comment:
This could be simplified a bit:
```suggestion
    public Map<Object, CompletableFuture<Object>> get(String cacheName) {
        return synchronousGetLocks.computeIfAbsent(cacheName, k -> new ConcurrentHashMap<>());
    }
```

Reply:
thanks!